src/components/common/StyledButton/index.js

```diff
@@ -11,25 +11,22 @@ const buttonColor = {
   gradient: "gradientBg",
   black: "blackBg",
   transparent: "transparentBg",
+  red: "red",
 };
 
-const StyledButton = props => {
+const StyledButton = ({ disabled, onClick, type, iconClass, btnText }) => {
   const classes = useStyles();
 
   return (
-    <Button
-      className={clsx(classes.styledButton, classes[buttonColor[props.type]])}
-      disabled={props.disabled}
-      onClick={props.onClick}
-    >
-      <Icon className={props.iconClass} />
-      {props.btnText}
+    <Button className={clsx(classes.styledButton, classes[buttonColor[type]])} disabled={disabled} onClick={onClick}>
+      {iconClass ? <Icon className={iconClass} /> : null}
```

Comment:
Is it possible to handle this check inside `Icon` component?

Reply:
[done](https://github.com/singnet/refactored-dapp/pull/17/commits/a9aeb169238bc86f6201495bdbb41c8a8c71cd5c)